src/msg/async/Stack.h

```diff
@@ -203,9 +203,28 @@ enum {
   l_msgr_send_messages_queue_lat,
   l_msgr_handle_ack_lat,
 
+  l_msgr_recv_encrypted_bytes_mon_cluster,
+  l_msgr_recv_encrypted_bytes_mon_service,
+  l_msgr_recv_encrypted_bytes_mon_client,
+  l_msgr_recv_encrypted_bytes_cluster,
+  l_msgr_recv_encrypted_bytes_service,
+  l_msgr_recv_encrypted_bytes_client,
+
+  l_msgr_send_encrypted_bytes_mon_cluster,
+  l_msgr_send_encrypted_bytes_mon_service,
```

Comment:
The delineation here has always been confusing: e.g. mon/mgr -> osd connection would be incrementing `mon_service` counters, whereas osd -> mon/mgr connection would be incrementing `mon_cluster` counters.  Do we really expect a user to make sense of this? (very much related to the use case question)

Reply:
Indeed, this is a pure pain and the conn types are a complex beast. These couters just follow as I'm worried that an attempt to simplify would introduce even more complexity.

The target audience are security folks / developers who want to ensure every single conn type works as expected when it comes to crypting its traffic.

Reply:
Perhaps a comment would be useful (for this particular feature users are fine with reading the source code, I think).

Reply:
I'm not convinced.  Someone who is really security-conscious probably wouldn't trust some random counters with very confusing nomenclature and resort to examining packet captures anyway.  Someone who is performing a quick check to see whether the cluster is configured as expected would probably be satisfied by a pair of global `encrypted_bytes` counters:
- for a default-configured cluster, where only monitor traffic is encrypted, they would check that `bytes == encrypted_bytes` on all monitors
- for an all-secure cluster, where all traffic is encrypted, they would check that `bytes == encrypted_bytes` on all daemons
- for an all-plain cluster, they would check that `encrypted_bytes == 0` on all daemons

While the existing configuration options allow for additional scenarios falling "between" these three, they don't make a lot of sense and I doubt anyone would ever go there because it's very hard to make sense of them.

Reply:
> Someone who is really security-conscious probably wouldn't trust some random counters with very confusing nomenclature and resort to examining packet captures anyway

Well, I'm afraid that taking a look on every single frame sent even over a day (which still might be not enough to catch a corner case) would be so labor-intensive that learning our `ConnenctionType` complexity looks like an acceptable cost ;-). I'm also worries that, during a security audit, it is not so important what is truly being used by people but what was declared usable :-(.

However, I think we both agree this complexity lacks a good justification and, probably, we can cut by half the number of types. The question remaining after the CDM is how to tackle that. I put the first commit (the very simplified version) to be able to send it separately. Maybe this is what should be done now.

Reply:
> I put the first commit (the very simplified version) to be able to send it separately. Maybe this is what should be done now.

I think it would be sufficient for now -- if someone's use case is not served by a simple pair of `encrypted_bytes` counters, I'm all ears.  But ultimately this is up to you and @neha-ojha, of course.